### ISSUE TITLE
Unexpected allocation in sort! of Int64 view

### ISSUE THREAD

**Issue by rickbeeloo** [JuliaLang/julia#47152]
Hey!

Allocations are made when sorting a view of an `Int64` array (tested on `1.7.3` and `1.8.0`), as discussed [here](https://discourse.julialang.org/t/allocation-when-sorting-a-view/86558/5?u=codegodz) on discourse.  @mkoculak spotted that such allocations do not occur for `Float64`:

```
x = [2,1,10,15,20]
y = [2.0,1.0,10.0,15.0,20.0]

@btime sort!(view($x,2:4))
33.898 ns (2 allocations: 96 bytes)

@btime sort!(view($y,2:4))
19.645 ns (0 allocations: 0 bytes)
```
I guess allocations should also not be required for `Int64`?

**Comment by LilithHafner** [comment]
This is already fixed in 1.9 (release candidates should be available soon).

1.8.1
```
julia> x = [2,1,10,15,20];

julia> y = [2.0,1.0,10.0,15.0,20.0];

julia> @btime sort!(view($x,2:4));
  73.804 ns (2 allocations: 96 bytes)

julia> @btime sort!(view($y,2:4));
  22.630 ns (0 allocations: 0 bytes)
```
1.9.0-DEV
```
julia> x = [2,1,10,15,20];

julia> y = [2.0,1.0,10.0,15.0,20.0];

julia> @btime sort!(view($x,2:4));
  15.062 ns (0 allocations: 0 bytes)

julia> @btime sort!(view($y,2:4));
  16.647 ns (0 allocations: 0 bytes)
```

**Comment by rickbeeloo** [comment]
Quite some speed improvement too! Thanks! :)
